llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -1908,8 +2032,12 @@ PackOp::reifyResultShapes(OpBuilder &builder,
     };
     // If we are dealing with a tiled dimension compose the map otherwise
     // return the dimension extracted with `memref.dim`.
+    // TODO(hanchung): Should bind inner size to inner_tiles if they are
+    // dynamic.
     OpFoldResult dimBound =
-        getDim(builder, getOperation()->getLoc(), getOutput(), dimIdx);
+        dimIdx < getInputRank()
+            ? getDim(builder, getOperation()->getLoc(), getInput(), dimIdx)
+            : getDim(builder, getOperation()->getLoc(), getOutput(), dimIdx);
```

Comment:
Found a bug when revisiting lit tests.

@MaheshRavishankar @chelini FYI, the outer dims of output should be `input[i] / tile[i]`, not `output[i] / tile[i]`.

I think we should get inner dim sizes by looking into inner_tiles. In this case, we don't have to create a redundant memref.dim (or tensor.dim) op.